### PR TITLE
Feature/participant demographics

### PR DESCRIPTION
**Pull request by maartenuni**
This pull request adds the following features:

1. Adds a submenu under the participants "header"
2. Provides users a histogram of overview of ages of the partiicpants in the database, also in order to help planning for a new experiment.
3. Create a custom command for mangage.py to simulate a recruitement of babies in a specific range
4. Create a custom command to undo point 3

### CODE REVIEW

File: participants/simulate.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        dyslexic_parent = _generate_parent_dylexia()
+        multilingual = _generate_parent_dylexia()
+        pweeks = int(round(random.normalvariate(WEEK_MU, WEEK_SIGMA)))
+        pdays = pweeks * 7 + random.choice(list(range(7)))
+        birth_weight = random.normalvariate(3500, 1000)
+        phonenumber = DEFAULT_PHONE
+        days_add = int(round((random.random() - .5) * day_range))
+        birth_date = date + timedelta(days=days_add)
+        gender = random.choice("FM")
+        city = "".join(reversed("Utrecht-City"))
```
Comment (bbonf) [inline participants/simulate.py:70]: This I would put in a constant

**Review by bbonf — COMMENTED** [review]
Matplotlib is also missing from requirements.in

File: participants/views.py [hunk 2]
```diff
@@ -126,3 +129,21 @@ def participant(self):
...
+
+class ParticipantsDemographicsView(generic.TemplateView,
+                                   braces.LoginRequiredMixin):
+
+    template_name = 'participants/demographics.html'
+
+    def get_context_data(self, **kwargs):
+        """Add path to image data to context"""
+        context = super().get_context_data(**kwargs)
+        context['png_demographics'] = reverse('png')
```
[the remark this answers is not in the document]
Comment (maartenuni) [reply participants/views.py:142]: No, didn't know that template internals had it's own way to reverse a path. I'll remove this.

File: participants/graphs.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    num_bins= max(delta_months) - min(delta_months) + 10
+    
+    ax.hist(delta_months, bins=num_bins)
+    ax.set_xlabel("Age in months")
+    ax.set_ylabel("Number of observations")
+
+    payload = BytesIO()
+
+    color = (9/10, 9/10, 9/10)
```
[the remark this answers is not in the document]
Comment (maartenuni) [reply participants/graphs.py:39]: I've made the background visible so I can see it, this way it aligns with the rest of the page. In new commit I'll make the background transparent and then we can see what we like better.

File: participants/urls.py [hunk 4]
```diff
@@ -2,12 +2,16 @@
 
 from .views import ParticipantDeleteView, ParticipantDetailView, \
     ParticipantSpecificCriteriaUpdateView, \
-    ParticipantUpdateView, ParticipantsHomeView
+    ParticipantUpdateView, ParticipantsHomeView, \
+    ParticipantsDemographicsView, render_demograhpics
 
 app_name = 'participants'
 
+DEMO_PATH = "demographics/"
```
[the remark this answers is not in the document]
Comment (maartenuni) [reply participants/urls.py:10]: I'll remove it

File: participants/simulate.py (hunk 1 above)
Comment (maartenuni) [reply participants/simulate.py]: Done

File: participants/simulate.py [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+                sex=gender,
+                email_subscription=random.choice([True, False]),
+                birth_weight=birth_weight,
+                pregnancy_weeks = pweeks,
+                pregnancy_days = pdays,
+                parent_name = "parent_" + name,
+                city = city
+                )
+
+        pp.save()
```
[the remark this answers is not in the document]
Comment (maartenuni) [reply participants/simulate.py]: Your right, fixed

File: participants/graphs.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    qset = Participant.objects.all()
+
+    today = dt.date.today()
+    delta_months = [diff_months(today, pp.birth_date) for pp in qset]
+
+    num_bins= max(delta_months) - min(delta_months) + 10
+    
+    ax.hist(delta_months, bins=num_bins)
+    ax.set_xlabel("Age in months")
+    ax.set_ylabel("Number of observations")
```
[the remark this answers is not in the document]
Comment (maartenuni) [reply participants/graphs.py:42]: There is a paragraph above the graph, perhaps we can add some extra info there or somekind of note underneath the graph, describing what it displays.

**Review by bbonf — APPROVED** [review]

**Pull request merged**